src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -569,6 +584,114 @@ FullQName AdvertiserMinMdns::GetCommisioningTextEntries(const CommissionAdvertis
     }
 }
 
+bool CanAdvertiseOn(chip::Inet::InterfaceAddressIterator & current)
+{
+    if (!current.IsUp())
+    {
+        return false;
+    }
+
+    if (!current.SupportsMulticast())
+    {
+        return false;
+    }
+
+    char name[chip::Inet::InterfaceIterator::kMaxIfNameLength];
+    if (current.GetInterfaceName(name, sizeof(name)) != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get interface name - will not advertise on this interface");
+        return false;
+    }
+
+    if (strncmp(name, "lo", 2) == 0)
+    {
+        // Skip advertisement on what looks like a local loopback interface
+        return false;
+    }
```

Comment:
We have this logic duplicated in a few places.  I know these are different iterator classes, but it might be worth having a template function to coalesce the logic even if it won't actually share the code in practice...

Reply:
Deduplicated within this file only for now - hard to use templates because even thoug the methods are identical (is up, isMulticast, getinterfacename) the iterator types are different.

Reply:
Right, I just meant within this file...